### ISSUE TITLE
fail2ban-regex support for .local files?

### ISSUE THREAD

**Issue by hlein** [fail2ban/fail2ban#954]
For real filter .conf files, a .local file can override specific settings, etc, which is great.

But, fail2ban-regex does not appear to look for a .local version of the regex filename, or have any way to explicitly provide one.  This diminishes fail2ban-regex's ability to be used to test or predict the behavior of fail2ban with a set of filter.d/foo{.conf,.local} files.

**Comment by yarikoptic** [comment]
On Wed, 11 Feb 2015, hlein wrote:

>    For real filter .conf files, a .local file can override specific settings,
>    etc, which is great.

>    But, fail2ban-regex does not appear to look for a .local version of the
>    regex filename, or have any way to explicitly provide one. This diminishes
>    fail2ban-regex's ability to be used to test or predict the behavior of
>    fail2ban with a set of filter.d/foo{.conf,.local} files.

hm.... just confirming that issue exists:

(git)hopa:~deb/fail2ban[debian]git
$> ls -l $PWD/config/filter.d/sshd.local
-rw------- 1 yoh yoh 1114 Feb 11 23:40 /home/yoh/deb/gits/fail2ban/config/filter.d/sshd.local

$> PYTHONPATH=$PWD bin/fail2ban-regex -l heavydebug fail2ban/tests/files/logs/sshd $PWD/config/filter.d/sshd.conf

Running tests
=============

Use   failregex file : /home/yoh/deb/gits/fail2ban/config/filter.d/sshd.conf
D:   Loading files: ['/home/yoh/deb/gits/fail2ban/config/filter.d/sshd.conf']
D:     Reading file: /home/yoh/deb/gits/fail2ban/config/filter.d/sshd.conf
D:   Loading files: ['/home/yoh/deb/gits/fail2ban/config/filter.d/common.conf']
D:     Reading file: /home/yoh/deb/gits/fail2ban/config/filter.d/common.conf
D:   Loading files: ['/home/yoh/deb/gits/fail2ban/config/filter.d/common.local']
D:     Reading file: /home/yoh/deb/gits/fail2ban/config/filter.d/common.local
D:   Loading files: ['/home/yoh/deb/gits/fail2ban/config/filter.d/common.conf', '/home/yoh/deb/gits/fail2ban/config/filter.d/sshd.conf']
D:   Loading files: ['/home/yoh/deb/gits/fail2ban/config/filter.d/common.conf']
D:     Reading file: /home/yoh/deb/gits/fail2ban/config/filter.d/common.conf
D:   Loading files: ['/home/yoh/deb/gits/fail2ban/config/filter.d/sshd.conf']
D:     Reading file: /home/yoh/deb/gits/fail2ban/config/filter.d/sshd.conf
D: Set maxlines = 10
Use         maxlines : 10
Use         log file : fail2ban/tests/files/logs/sshd
Use         encoding : UTF-8


so no sshd.local gets loaded for some reason although imho should

-- 
Yaroslav O. Halchenko, Ph.D.
http://neuro.debian.net http://www.pymvpa.org http://www.fail2ban.org
Research Scientist,            Psychological and Brain Sciences Dept.
Dartmouth College, 419 Moore Hall, Hinman Box 6207, Hanover, NH 03755
Phone: +1 (603) 646-9834                       Fax: +1 (603) 646-1419
WWW:   http://www.linkedin.com/in/yarik

**Comment by sebres** [comment]
fixed in #955 

`readexplicit` does not support completely read of all config files (in includes only).
But standard `read` expected instead of filename a filtername, and it should be relative base directory and inside `filter.d`.

**Comment by sebres** [comment]
```bash
PYTHONPATH=. bin/fail2ban-regex -l heavydebug 'test' $(pwd)/config/filter.d/sshd
```

```
Running tests
=============

Use   failregex filter file : sshd, basedir: /home/fail2ban.master/config
D: Loading configs for filter.d/sshd under /home/fail2ban.master/config
D: Reading configs for filter.d/sshd under /home/fail2ban.master/config
D: Reading config files: /home/fail2ban.master/config/filter.d/sshd.conf, /home/fail2ban.master/config/filter.d/sshd.local
D:   Loading files: ['/home/fail2ban.master/config/filter.d/sshd.conf']
D:     Reading file: /home/fail2ban.master/config/filter.d/sshd.conf
D:   Loading files: ['/home/fail2ban.master/config/filter.d/common.conf']
D:     Reading file: /home/fail2ban.master/config/filter.d/common.conf
D:   Loading files: ['/home/fail2ban.master/config/filter.d/common.local']
D:     Reading file: /home/fail2ban.master/config/filter.d/common.local
D:   Loading files: ['/home/fail2ban.master/config/filter.d/sshd.local']
D:     Reading file: /home/fail2ban.master/config/filter.d/sshd.local
D:   Loading files: ['/home/fail2ban.master/config/filter.d/common.conf', '/home/fail2ban.master/config/filter.d/sshd.conf', '/home/fail2ban.master/config/filter.d/sshd.local']
D:     Shared file: /home/fail2ban.master/config/filter.d/common.conf
D:     Shared file: /home/fail2ban.master/config/filter.d/sshd.conf
D:     Shared file: /home/fail2ban.master/config/filter.d/sshd.local
D: Set maxlines = 10
Use         maxlines : 10
Use      single line : test
...
```
</spoiler>
